### PR TITLE
TDI-36794: Change custom CSV writing to a standard one

### PR DESCRIPTION
**Pull request by fhuaulme**
* Use already present CSV writer instead of output stream write calls.
* Unit test for line feed handling.

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/80/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/80/)

**Comment by sgandon** [comment]
I just changed the IllegalArgumentException into a TalendRuntimeException.
Waiting for the merge test to pass again then I merge it into master and cherry-pick it on the maintenance/0.13 branch.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/84/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/84/)

File: components-dataprep/src/main/java/org/talend/components/dataprep/runtime/DataSetWriter.java [hunk 1]
```diff
@@ -78,8 +86,11 @@ public void open(String uId) throws IOException {
             outputStream = connectionHandler.write(DataPrepOutputModes.LiveDataset);
             break;
         default:
-            break;
+            throw new TalendRuntimeException(CommonErrorCodes.UNEXPECTED_EXCEPTION,
+                    ExceptionContext.build().put("message", "Mode '" + mode + "' is not supported."));
         }
+        // Use a CSV writer iso. output stream
+        writer = new CsvWriter(new OutputStreamWriter(outputStream), ';');
```
Comment (wwang-talend) [inline components-dataprep/src/main/java/org/talend/components/dataprep/runtime/DataSetWriter.java]: use ";" instead of ","?

**Pull request merged**